util/movement-types/src/lib.rs

```diff
@@ -138,4 +138,32 @@ impl Block {
         self.transactions.push(transaction);
     }
 
+}
+
+#[derive(Serialize, Deserialize, Clone, Default, Debug, PartialEq, Eq, Hash, PartialOrd, Ord)]
+pub struct Commitment(pub Vec<u8>);
+
+impl Commitment {
+    pub fn test() -> Self {
+        Self(vec![0])
+    }
+}
+
+impl From<Vec<u8>> for Commitment {
+    fn from(data : Vec<u8>) -> Self {
+        Self(data)
+    }
+}
+
+impl From<Commitment> for Vec<u8> {
+    fn from(commitment : Commitment) -> Vec<u8> {
+        commitment.0
+    }
+}
+
+#[derive(Serialize, Deserialize, Clone, Default, Debug, PartialEq, Eq, Hash, PartialOrd, Ord)]
+pub struct BlockCommitment {
+    pub height : u64,
+    pub block_id : Id,
+    pub commitment : Commitment,
```

Comment:
I think we need e2e tests here to call the MCR contract https://github.com/movementlabsxyz/movement/pull/48

We need these tests before we can be confident.

Reply:
https://github.com/movementlabsxyz/movement/pull/58#issuecomment-2107390306